core/arch/arm/plat-nuvoton/main.c

```diff
@@ -0,0 +1,82 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2016-2023, Linaro Limited
+ * Copyright (c) 2014-2023, STMicroelectronics International N.V.
+ * Copyright (C) 2022-2023 Nuvoton Ltd.
+ */
+
+#include <console.h>
+#include <drivers/gic.h>
+#include <drivers/ns16550.h>
+#include <kernel/boot.h>
+#include <kernel/interrupt.h>
+#include <kernel/linker.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <trace.h>
+
+#if defined(CFG_NPCM_DEBUG)
+#define COLOR_NORMAL	"\x1B[0m"
+#define COLOR_RED	"\x1B[31m"
+#define COLOR_GREEN	"\x1B[32m"
+#define COLOR_YELLOW	"\x1B[33m"
+#define COLOR_BLUE	"\x1B[34m"
+#define COLOR_MAGENTA	"\x1B[35m"
+#define COLOR_CYAN	"\x1B[36m"
+#define COLOR_WHITE	"\x1B[37m"
+#endif /* CFG_NPCM_DEBUG */
+
+static struct gic_data gic_data __nex_bss;
+static struct ns16550_data console_data __nex_bss;
+
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, UART_REG_SIZE);
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GICD_BASE, GIC_DIST_REG_SIZE);
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GICC_BASE, GIC_DIST_REG_SIZE);
+
+register_ddr(DRAM0_BASE, DRAM0_SIZE);
+
+#if defined(CFG_NPCM_DEBUG)
+static void print_version(void)
+{
+	IMSG(COLOR_MAGENTA);
+	IMSG(">================================================");
+	IMSG("OP-TEE OS Version %s", core_v_str);
+	IMSG(">================================================");
+	IMSG(COLOR_NORMAL);
+}
+#endif /* CFG_NPCM_DEBUG */
+
+void main_init_gic(void)
+{
+#if defined(CFG_NPCM_DEBUG)
+		print_version();
+#endif /* CFG_NPCM_DEBUG */
+
+#if defined(CFG_WITH_ARM_TRUSTED_FW)
+	/* On ARMv8, GIC configuration is initialized in ARM-TF */
+	gic_init_base_addr(&gic_data, GIC_BASE + GICC_OFFSET,
+			   GIC_BASE + GICD_OFFSET);
+#else
+	/* Initialize GIC */
+	gic_init(&gic_data, GIC_BASE + GICC_OFFSET, GIC_BASE + GICD_OFFSET);
```

Comment:
I removed CFG_WITH_ARM_TRUSTED_FW check, because the result is the same.

As for print_version, I leaved IF DEFINED because otherwise I received a compilation error:
error: implicit declaration of function 'print_version' [-Werror=implicit-function-declaration]
   55 |                 print_version();